lib/agenda.js

```diff
@@ -556,18 +555,7 @@ function processJobs(extraJob) {
       return;
     }
 
-    var criteria = {
-      _id: job.attrs._id,
-      lockedAt: null,
-      nextRunAt: job.attrs.nextRunAt,
-      disabled: { $ne: true }
-    };
-
-    var sort = {};
-    var update = { $set: { lockedAt: now } };
-    var options = { new: true };
-
-    self._collection.findAndModify(criteria, sort, update, options, function(err, resp) {
+    self._collection.findAndModify({ _id: job.attrs._id, lockedAt: null, disabled: { $ne: true } }, {}, { $set: { lockedAt: now } }, { new: true }, function(err, resp) {
```

Comment:
Looks like you reverted most of the commits from past weeks in this PR (ex: https://github.com/rschmukler/agenda/commit/d13ecfdbf630746a39c95dbb1287e16b4f02a2ad)